### ISSUE TITLE
Prevent overfitting on speakers

### ISSUE THREAD

**Issue by smolendawid** [danijel3/ClarinStudioKaldi#1]
In the paper it is said that
> The corpus consists of 317 speakers recorded in 554 sessions

and

> 56 random sessions were chosen to be in the test set, and the remaining 499 session were stored in the test [mistake, it is train] set. 

Does it mean that the overfitting in terms of speakers was possible? I mean the same speaker can be in train and test set. The results don't seem reliable...

Is there any information about speaker ID in database? Then it could be possible to test the baseline system truly

**Comment by danijel3** [comment]
Yes, that is correct. Test set was chosen randomly, so same person can be in both.

In speech recognition we are not attempting to differentiate speakers (we try to minimize the speaker variability in data) and detecting speakers is not something we look for explicitly, but I can see your point. If we want to be completely accurate, we should have accurate speaker IDs.

I will keep this issue open until we create accurate speaker IDs. It shouldn't be too difficult to do manually, since the same speakers are always in sequential session. In essence, we just need to check if the two sessions in a sequence have the same speaker or not.

**Comment by smolendawid** [comment]
Thank you for the response. Sorry for the sharp tone - in the field I come from (medical electrical signal analysis), overfitting occurs in many senses, which leads to a lot of misunderstandings and disappointments. 

I think it will be very time-consuming to validate the system again, however, thank you for the comprehension

**Comment by danijel3** [comment]
It's no problem - that's what issues are for. It shouldn't be too difficult. I have to listen over 550 ~1 second audio fragments. Shouldn't take more than an hour. Just need some more time in general.

**Comment by danijel3** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Prevent overfitting on speakers](https://trello.com/c/hFIvaoPA/14-prevent-overfitting-on-speakers)

**Comment by danijel3** [comment]
Fixed in 3a47f1c582594148b1bc70b8d55f7035ff97e041
